### PR TITLE
feat: Implementation of country code Drop-Down 

### LINKED ISSUE #2578: Implementation of country code Drop-Down
**Is your feature request related to a problem? Please describe.**

No, I would like to implement a dropdown for country code for mobile numbers with their flag .

**Describe the solution you'd like**

Implement the drop-down

**Additional context**

Would like to work on it.

### PR DESCRIPTION
**Pull request by snitin315**
Fixes #2578 

Screenshot from 2020-02-16 15-26-56

### CODE REVIEW

File: app/utils/dictionary/demography.ts [hunk 1]
```diff
@@ -4,1139 +4,1423 @@
...
+    name        : 'Puerto Rico',
+    code        : 'PR',
+    dialingCode : '+1 787'
   },
   {
-    name : 'Palestinian Territory, Occupied',
-    code : 'PS'
+    name        : 'Palestinian Territory, Occupied',
+    code        : 'PS',
+    dialingCode : '+970',
```
Comment (niranjan94) [inline app/utils/dictionary/demography.ts]: Codacy Issue found: [Unexpected trailing comma.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=5010957)

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 3
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=5010957&bid=16527768)

File: app/utils/dictionary/demography.ts [hunk 2]
```diff
@@ -4,1139 +4,1423 @@
...
-    name : 'Brunei Darussalam',
-    code : 'BN'
+    name        : 'Brunei Darussalam',
+    code        : 'BN',
+    dialingCode : '+673'
   },
-  {
-    name : 'Bolivia',
-    code : 'BO'
+  {
```
Comment (niranjan94) [inline app/utils/dictionary/demography.ts]: Codacy Issue found: [Trailing spaces not allowed.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=5010957)

File: app/utils/dictionary/demography.ts [hunk 3]
```diff
@@ -4,1139 +4,1423 @@
...
+    name        : 'Monaco',
+    code        : 'MC',
+    dialingCode : '+377'
   },
   {
-    name : 'Moldova',
-    code : 'MD'
+    name        : 'Moldova',
+    code        : 'MD',
+    dialingCode : '+373'
```
Comment (niranjan94) [inline app/utils/dictionary/demography.ts]: Codacy Issue found: [Trailing spaces not allowed.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=5010957)

File: app/utils/dictionary/demography.ts [hunk 4]
```diff
@@ -4,1139 +4,1423 @@
  */
 export const countries = [
   {
-    name : 'Ascension Island',
-    code : 'AC'
+    name        : 'Ascension Island',
+    code        : 'AC',
+    dialingCode : '+247'
```
Comment (kushthedude) [inline app/utils/dictionary/demography.ts]: Use ~dialingCode~ phone

File: app/utils/dictionary/demography.ts [hunk 5]
```diff
@@ -4,1139 +4,1423 @@
...
-    code : 'XK'
+    name  : 'Kosovo',
+    code  : 'XK',
+    phone : '+383'
   },
   {
-    name : 'Yemen',
-    code : 'YD'
+    name        : 'Yemen',
+    code        : 'YD',
```
Comment (niranjan94) [inline app/utils/dictionary/demography.ts]: Codacy Issue found: [Extra space after key 'code'.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=5010957)

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=5010957&bid=16527768)

File: app/utils/dictionary/demography.ts [hunk 6]
```diff
@@ -4,1139 +4,1423 @@
...
-    name : 'Kosovo',
-    code : 'XK'
+    name  : 'Kosovo',
+    code  : 'XK',
+    phone : '+383'
   },
   {
-    name : 'Yemen',
-    code : 'YD'
+    name        : 'Yemen',
```
Comment (niranjan94) [inline app/utils/dictionary/demography.ts]: Codacy Issue found: [Extra space after key 'name'.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=5010957)

**Comment by snitin315** [comment]
@kushthedude there are some country names present like `European Union` for which it's complicated to have phone , should i leave them empty?

Screenshot from 2020-02-16 15-05-28

File: app/components/forms/user-profile-form.js [hunk 7]
```diff
@@ -57,6 +59,8 @@ export default Component.extend(FormMixin, {
     };
   },
 
+  countries   : orderBy(countries, 'name'),
```
Comment (niranjan94) [inline app/components/forms/user-profile-form.js]: Codacy Issue found: [Extra space after key 'countries'.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=5010957)

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=5010957&bid=16527768)

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=5010957&bid=16527768)

File: app/components/forms/user-profile-form.js [hunk 8]
```diff
@@ -57,6 +59,8 @@ export default Component.extend(FormMixin, {
     };
   },
 
+  countries : orderBy(countries, 'name'),
```
Comment (niranjan94) [inline app/components/forms/user-profile-form.js]: Codacy Issue found: [Extra space after key 'countries'.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=5010957)

**Comment by snitin315** [comment]
@kushthedude right now I have implemented it in `user-profile-form` , please check if I'm approaching th right way.

**Comment by kushthedude** [comment]
Wrong, how will the code get stored with phone number. Code should be a attribute of phone number and ultimately will be stored with phone number.

**Comment by snitin315** [comment]
@kushthedude I think we should create a new widget for `phone-input` like there is for `link-input` as phone field is used multiple times in app.

**Comment by kushthedude** [comment]
Isnt only user-profile/speaker form contains it?

**Comment by snitin315** [comment]
[https://github.com/fossasia/open-event-frontend/search?q=phone&unscoped_q=phone](https://github.com/fossasia/open-event-frontend/search?q=phone&unscoped_q=phone)

account/contact-info ,  user-profile-form etc.

**Comment by snitin315** [comment]
> Code should be a attribute of phone number and ultimately will be stored with phone number.

@kushthedude I am not sure how to do this can you point me to some refernce?

**Comment by iamareebjamal** [comment]
@CosmicCoder96 He has done that only

**Comment by snitin315** [comment]
>why the .ts file when all others are js, 

@CosmicCoder96 recently an issue was raised to migrate the project to typescript and hence most of the utils files were converted to `.ts` including `country dictioinary` as well.

**Comment by CosmicCoder96** [comment]
Oh apologies, went through the project after a really long time, wasn’t
aware of that.

**Comment by CosmicCoder96** [comment]
> I am not sure how to do this can you point me to some refernce?
Check how social media URL input widget works.

**Comment by kushthedude** [comment]
Superseded by #4572

**Pull request closed without merging**